docs/schemas/trace-analytics/readme.md

```diff
@@ -0,0 +1,91 @@
+* ***Forwards compatibility*** - features built on version 1.x of the schema **will not break** if data from a **later** 1.x schema version is used.
+
+### Read-compatibility examples
+
+1. A plugin built on schema 1.1 but consuming schema 1.2 data will function 100% without issue
+2. A plugin built on schema 1.2 but consuming schema 1.1 data will continue to function, however some features might be degraded
+3. A plugin built on schema 1.0 but consuming schema 2.0 data is not guaranteed to function
+4. A plugin built on schema 2.0 but consuming schema 1.0 data is not guaranteed to function
+
+### Write-compatibility examples
+
+1. A writer built on schema 1.2 but writing to a cluster containing schema 1.1 data will succeed
+2. A writer built on schema 1.1 but writing to a cluster containing schema 1.2 data will succeed
+3. A writer built on schema 1.0 but writing to a cluster containing schema 2.0 data will succeed
+4. A writer built on schema 2.0 but writing to a cluster containing schema 1.0 data will succeed
+
+### Handling minor version updates
+
+Minor version updates will occur as new fields are needed to support new Trace Analytics features. The steps to update a schema minor version are to:
+
+1. Ensure both TA and DP owners are aligned with requirements
+2. Update the schema in the schema repo
+    1. Add new fields to the schema JSON
+    2. Increment the `version` field of the schema by 1
+    3. Update the documentation to describe the new fields
+3. Add test data to the TA plugin test suite
+    1. Don't update existing data in place, instead add new data following the new schema version. The test suite is expected to pass with a range of minor versions being tested at the same time.
+4. Update DP to start emitting documents following the new schema version
+5. Add new features to the TA plugin utilizing the new data
+
+### Handling Major version updates
+
+As schema owners, we will do our best to avoid introducing major version changes. However as our schemas are heavily tied to the OpenTelemetry spec, there is always the risk of an upstream backwards-incompatible change requiring a major version increase.
+
+Due to the potentially disjointed release schedules of both OpenSearch and the managed offering, we need to ensure that rolling out a major version change is carefully planned.
+
+A typical migration plan will first make Data Prepper artifacts available so that users can start ingesting their data to the new index. To prevent data loss during the migration period, users can be encouraged to simultaneously write to both the old and new indexes. This can be done by either running both old and new versions of Data Prepper side-by-side, or perhaps Data Prepper itself can be updated to write to dual indexes (TODO). Once users have the ability to write to the new index, Trace Analytics plugin updates will be made available which make use of the new major version index.
```

Comment:
> or perhaps Data Prepper itself can be updated to write to dual indexes (TODO)

Suggest take down this part until there is clarity on what Data Prepper can do.

Reply:
> To prevent data loss during the migration period, users can be encouraged to simultaneously write to both the old and new indexes. 

QUES: not understand this part well. Why there is a data loss and how simultaneously write to both indexes resolve it?

Reply:
~~To prevent data loss~~ To minimize analytics downtime. If a user doesn't care to reindex old data, then they'll simply upgrade from Dashboard v1 to Dashboard v2. Dual writing to both v1 and v2 indexes allows for new data to be viewed both before and after the upgrade.

Will clean this section up a bit.